docs/proposals/20240807-in-place-updates.md

```diff
@@ -395,6 +403,441 @@ However, in-place updates might cause Nodes to become unhealthy while the update
+
+```json
+{
+    "machine": {
+        "name": "kcp-1-hfg374h",
+        "namespace": "default",
+    }
+}
+```
+
+The `kcp-version-upgrade` which has tracked the upgrade process reported by the agent, responds:
+
+```json
+{
+    "error": null,
+    "status": "Done"
+}
+```
+
+The Machine controller then removes the updater from the spec and marks the machine as up to date:
+
+```diff
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: Machine
+metadata:
+  name: kcp-1-hfg374h
+spec:
+  version: v1.31.0
+  bootstrap:
+    configRef:
+      apiVersion: bootstrap.cluster.x-k8s.io/v1beta1
+      kind: KubeadmConfig
+      name: kcp-1-hfg374h-flkf3
+      uid: ddab8525-bb36-4a86-81e9-ef3eeeb33e18
+-  updaters:
+-  - kcp-version-upgrade
+status:
+  conditions:
+- - lastTransitionTime: "2024-12-31T23:50:00Z"
+-   status: "False"
++ - lastTransitionTime: "2024-12-31T23:59:59Z"
++   status: "True"
+    type: UpToDate
+```
+
+On the next KCP reconciliation, it detects that this machine doesn't differ anymore from the desired state, picking the next and repeating the same process.
+
+This process is repeated a third time with the last KCP machine, finally marking the KCP object as up to date.
+
+#### Update worker node memory
+
+```mermaid
+sequenceDiagram
+    participant Operator
+
+    box Management Cluster
+        participant apiserver as kube-api server
+        participant capi as  MD controller
+        participant msc as MachineSet Controller
+        participant mach as Machine Controller
+        participant hook as vSphere memory <br>update extension
+    end
+    
+
+    participant api as vSphere API
+
+    Operator->>+apiserver: Update memory field in<br> VsphereMachineTemplate
+    apiserver->>+capi: Notify changes
+    apiserver->>-Operator: OK
+    capi->>+hook: Can update [spec.memoryMiB]?
+        hook->>capi: I can update [spec.memoryMiB]
+    capi->>apiserver: Create new MachineSet
+    loop For all Machines
+        capi->>apiserver: Mark Machine with Update Plan<br> and move to new MachineSet
+        apiserver->>msc: Notify changes
+        msc->>apiserver: Update Machine's<br> spec.memoryMiB
+        apiserver->>mach: Notify changes
+        mach->>hook: Run update in<br> in Machine
+        hook->>mach: In progress
+        hook->>api: Update VM's memory
+        api->>hook: Done
+        mach->>hook: Run update in<br> in Machine
+        hook->>mach: Done
+        mach->>apiserver: Mark Machine as updated
+    end
+```
+
+The user starts the process by creating a new VSphereMachineTemplate with the updated `memoryMiB` value and updating the infrastructure template ref in the MachineDeployment:
+
+```diff
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: VSphereMachineTemplate
+metadata:
+  name: md-1-2
```

Comment:
The name should probably change here to indicate this is a new resource, not editing an existing one.